admin/grants.sql

```diff
@@ -680,4 +680,34 @@ GRANT USAGE, MONITOR
     TO DATABASE ROLE SYNAPSE_DATA_WAREHOUSE.SYNAPSE_RAW_ALL_DEVELOPER;
 GRANT USAGE, MONITOR
     ON SCHEMA SYNAPSE_DATA_WAREHOUSE_DEV.SYNAPSE_RAW
-    TO DATABASE ROLE SYNAPSE_DATA_WAREHOUSE_DEV.SYNAPSE_RAW_ALL_DEVELOPER;
\ No newline at end of file
+    TO DATABASE ROLE SYNAPSE_DATA_WAREHOUSE_DEV.SYNAPSE_RAW_ALL_DEVELOPER;
+
+-- Grant schema privileges for SYNAPSE_AGGREGATE schema
+-- SYNAPSE_DATA_WAREHOUSE
+GRANT OWNERSHIP
+	ON SCHEMA SYNAPSE_DATA_WAREHOUSE.SYNAPSE_AGGREGATE
+	TO DATABASE ROLE SYNAPSE_DATA_WAREHOUSE.SYNAPSE_AGGREGATE_ALL_ADMIN;
+GRANT USAGE, MONITOR
+    ON SCHEMA SYNAPSE_DATA_WAREHOUSE.SYNAPSE_AGGREGATE
+    TO DATABASE ROLE SYNAPSE_DATA_WAREHOUSE.SYNAPSE_AGGREGATE_ALL_ANALYST;
+GRANT USAGE, MONITOR
+    ON SCHEMA SYNAPSE_DATA_WAREHOUSE.SYNAPSE_AGGREGATE
+    TO DATABASE ROLE SYNAPSE_DATA_WAREHOUSE.SYNAPSE_AGGREGATE_ALL_DEVELOPER;
+GRANT OWNERSHIP
+	ON ALL TABLES
+	IN SCHEMA SYNAPSE_DATA_WAREHOUSE.SYNAPSE_AGGREGATE
+	TO DATABASE ROLE SYNAPSE_DATA_WAREHOUSE.SYNAPSE_AGGREGATE_ALL_ADMIN;
+-- SYNAPSE_DATA_WAREHOUSE_DEV
+GRANT OWNERSHIP
+    ON SCHEMA SYNAPSE_DATA_WAREHOUSE_DEV.SYNAPSE_AGGREGATE
+    TO DATABASE ROLE SYNAPSE_DATA_WAREHOUSE_DEV.SYNAPSE_AGGREGATE_ALL_ADMIN;
+GRANT USAGE, MONITOR
```

Comment:
I'm a bit confused. As this chart, `SYNAPSE_AGGREGATE_TABLE_READ` should have SELECT, REFERENCES privileges and inherited by  `SYNAPSE_AGGREGATE_ALL_ANALYST`. But I only saw it has privileges on future objects. Is this what we expected?

<img width="989" alt="Screenshot 2025-04-10 at 4 44 28 PM" src="https://github.com/user-attachments/assets/59615cdc-53ea-4ef7-94ad-f54fb21acb01" />

Reply:
Nice catch. I originally thought we wouldn't have any tables, but since we're creating a dummy table I need to grant table read privileges on that table explicitly in `admin/grants.sql`.

BTW, that chart is from the "conceptual" RBAC framework document. I wrote [a different document](https://sagebionetworks.jira.com/wiki/spaces/DPE/pages/3829006353/Synapse+Data+Warehouse+Role+Hierarchy#Role-Hierarchy) that has diagrams specifically for the Synapse data warehouse.

Reply:
Thanks for sharing. I will read the doc you shared.